ocs/ocs_feed.py

```diff
@@ -219,3 +222,26 @@ def publish_message(self, message, timestamp=None):
         else:
             # Publish message immediately
             self.agent.publish(self.address, (message, self.encoded()))
+
+    @staticmethod
+    def verify_message_data_type(message):
+        """Aggregated Feeds can only store certain types of data. Here we check
+        that the type of all data contained in a message's 'data' dictionary are
+        supported types.
+
+        Args:
+            message (dict):
+                Data to be published (see Feed.publish_message for details).
+
+        """
+        valid_types = (float, int)
+
+        for k, v in message['data'].items():
+            # multi-sample check
+            if isinstance(v, list):
+                if not all(isinstance(x, valid_types) for x in v):
+                    raise ValueError("message 'data' block contains invalid data type")
```

Comment:
I think this error message should be a bit more informative. Can we log the feed_name, block_name, data field, value, and type that's being passed? Or do you think that's unnecessary info?

Reply:
I did add the invalid types passed to the error. I debated on the other info. Sort of working off of examples in the Python standard library for TypeErrors. I think it's reasonable to expect some level of debugging from a user writing an Agent that hits this TypeError. I was also thinking of adding some debug level logging with the other requested info.